pkg/system/phase2_creating.go

```diff
@@ -812,30 +811,152 @@ func (r *Reconciler) ReconcileRootSecret() error {
 	}
 	r.setKMSConditionType(k.Type)
 
-	v, err := k.Get()
-	if err != nil {
-		r.Logger.Errorf("ReconcileRootSecret, KMS Get error %v", err)
+	conditionStatus := nbv1.ConditionKMSSync // default condition status
+	if err := k.Get(); err != nil {
+		if err != secrets.ErrInvalidSecretId {
+			// Unknown get error
+			r.Logger.Errorf("ReconcileRootSecret, KMS Get error %v", err)
+			r.setKMSConditionStatus(nbv1.ConditionKMSErrorRead)
+			return err
+		}
+
 		// the KMS root key was empty
 		// Initialize external KMS with a randomly generated key
-		if err == secrets.ErrInvalidSecretId {
-			r.SecretRootMasterKey = util.RandomBase64(32)
-			err := k.Set(r.SecretRootMasterKey)
-			if err != nil {
-				r.Logger.Errorf("ReconcileRootSecret, KMS Set error %v", err)
-				r.setKMSConditionStatus(nbv1.ConditionKMSErrorWrite)
-				return err
-			}
-			r.setKMSConditionStatus(nbv1.ConditionKMSInit)
-			return nil
+		err = k.Set(util.RandomBase64(32))
+		if err != nil {
+			r.Logger.Errorf("ReconcileRootSecret, KMS Set error %v", err)
+			r.setKMSConditionStatus(nbv1.ConditionKMSErrorWrite)
+			return err
 		}
-		// Unknown get error
-		r.setKMSConditionStatus(nbv1.ConditionKMSErrorRead)
+		// The key was set the first time - status init
+		conditionStatus = nbv1.ConditionKMSInit
+	}
+
+	// Set the value from KMS
+	err = k.Reconcile(r)
+	if err != nil {
+		r.Logger.Errorf("ReconcileRootSecret, KMS reconcile error %v", err)
+		r.setKMSConditionStatus(nbv1.ConditionKMSErrorSecretReconcile)
+		return err
+	}
+
+	r.setKMSConditionStatus(conditionStatus)
+
+	if err := r.ReconcileKeyRotation(); err != nil {
 		return err
 	}
+
+	return nil
+}
+
+// ReconcileKeyRotation checks if key rotation is enabled
+// if so creates a cron job to run every time set
+func (r *Reconciler) ReconcileKeyRotation() error {
+	r.Logger.Infof("ReconcileKeyRotation, KMS Starting")
+
+	if len(r.SecretRootMasterKey) > 0 {
+		// key rotation not supported by the KMS backend
+		r.Logger.Infof("ReconcileKeyRotation, KMS skip reconcile single master root mode")
+		return nil
+	}
+
+	keyManagementServiceSpec := r.NooBaa.Spec.Security.KeyManagementService
```

Comment:
I think we need both flags in case the user wants to set a schedule, but wants to disable it after some time. And for the option to stop key rotation totally. I think what I want to say is:
if enabled and schedule not set => default should be used.
if disable and schedule set or not set => the rotate will never run until will be set to enabled.
the default will be enabled, and schedule of once a month.

Reply:
I have two concerns with `EnableKeyRotation`:
- The possibility of inconsistent state if rotation is enabled by configuration but not supported by the backend.
- The fact that it is a new mandatory field, which would render existing CRs invalid and cause issues with upgrades.
wdyt?

Reply:
Re-added optional EnableKeyRotation, per [OCS operator PR]( https://github.com/red-hat-storage/ocs-operator/pull/1926)